components/_util/responsiveObserve.ts

```diff
@@ -12,6 +12,7 @@ export const responsiveMap: BreakpointMap = {
   lg: '(min-width: 992px)',
   xl: '(min-width: 1200px)',
   xxl: '(min-width: 1600px)',
+  xxxl: '(min-width: 2000px)',
```

Comment:
Too many break point.
This seems not a good design. Which should be modified to read from [token](https://github.com/ant-design/ant-design/blob/master/components/theme/util/alias.ts#L27) and set by ConfigProvider with `theme` instead.

Reply:
If that is the decision you make guys, I don't know if I'll be able to make the changes myself, I don't really understand what you're talking about, and still less how to implement it :/

Reply:
@yoyo837 
I'll try something, do I need to make something like https://github.com/ant-design/ant-design/blob/master/components/grid/style/index.tsx#L139 and line 150 ?

Reply:
As above mentioned. It should be able to reconfig break point with ConfigProvider:
```tsx
<ConfigProvider
  theme={{
    token: {
      screenXS: xxx, // Some new value
      ...
    },
  }}
>
</ConfigProvider>
```

For this, current break point not fully control by theme token. It still exist in some part outside: https://github.com/ant-design/ant-design/blob/77ea4038d169d033dcd4d29da1df9ab21b23ebb9/components/_util/responsiveObserve.ts#L8

What should do is modify up code to support consume ConfigProvider token.

Reply:
After reading more code, it seems that `genComponentStyleHook` is not what I need, I think that is something like the following is needed, but I don't where to get the token from 
```js
// 'fromToken' doesn't exist, it is just to show what I think I need 

const useResponsiveStyle = fromToken((token) => {
  return {
    'xs': `(max-width: ${token.screenXS}px)`,
    'sm': `(max-width: ${token.screenSM}px)`,
    'md': `(max-width: ${token.screenMD}px)`,
    'lg': `(max-width: ${token.screenLG}px)`,
    'xl': `(max-width: ${token.screenXL}px)`,
    'xxl': `(max-width: ${token.screenXXL}px)`,
    'xxxl': `(max-width: ${token.screenXXXL}px)`,
  };
});

export const responsiveMap: BreakpointMap = useResponsiveStyle();
```

I'm ok to continue on this PR to make this work, but I need some guidance please :)

Reply:
> As above mentioned. It should be able to reconfig break point with ConfigProvider:
> ...
> What should do is modify up code to support consume ConfigProvider token.

@zombieJ  That I got it now. But I don't find what method to use for, can you point me some place where that thing is done (or maybe some documentation about how to implement it) ?

Reply:
@zombieJ @yoyo837 

Is that something like this in `compoments/_util/responsiveObserve.ts` file ? 

```js
import { useToken } from '../../theme/internal';

const getResponsiveMap = (): BreakpointMap => {
  const [, token, ] = useToken();
  return {
    xs: `(max-width: ${token.screenXS}px)`,
    sm: `(max-width: ${token.screenSM}px)`,
    md: `(max-width: ${token.screenMD}px)`,
    lg: `(max-width: ${token.screenLG}px)`,
    xl: `(max-width: ${token.screenXL}px)`,
    xxl: `(max-width: ${token.screenXXL}px)`,
    xxxl: `(max-width: ${token.screenXXXL}px)`,
  } as BreakpointMap;
};

export const responsiveMap: BreakpointMap = getResponsiveMap();
```

Reply:
origin `compoments/_util/responsiveObserve.ts` export default with `responsiveObserve` object which do not have the ability to consume token context. It should change to hooks version:

```diff
  // compoments/_util/responsiveObserve.ts

- export default responsiveObserve;
+ export default function useResponsiveObserve() {
+   const [, token] = useToken();
+   // ...
+ }
```

And then adjust the code which use `responsiveObserve` to hooks usage.

Reply:
@zombieJ That would make a lots of changes, in many files

Don't my actual idea would fit, just make the responsiveMap a hook ? 

```js
const useResponsiveMap = (): BreakpointMap => {
  const [, token] = useToken();
  return {
    xs: `(max-width: ${token.screenXS}px)`,
    sm: `(max-width: ${token.screenSM}px)`,
    md: `(max-width: ${token.screenMD}px)`,
    lg: `(max-width: ${token.screenLG}px)`,
    xl: `(max-width: ${token.screenXL}px)`,
    xxl: `(max-width: ${token.screenXXL}px)`,
    xxxl: `(max-width: ${token.screenXXXL}px)`,
  } as BreakpointMap;
};
```

That only causes 2 changes in the same file 
```js
  unregister() {
    const responsiveMap: BreakpointMap = useResponsiveMap();
    Object.keys(responsiveMap).forEach((screen: Breakpoint) => {
...
  register() {
    const responsiveMap: BreakpointMap = useResponsiveMap();
    Object.keys(responsiveMap).forEach((screen: Breakpoint) => {
```

Reply:
@zombieJ I've done something , I let you see if that corresponds to your expectation

Basically, I've done this :

```diff
-   const token = ResponsiveObserve.subscribe((screen) => {
+  const responsiveObserve = useResponsiveObserve();
+  const token = responsiveObserve.subscribe((screen) => {
```